### DISCUSSION TITLE
How do I implement a view that follows a vehicle without shaking in the simulator window.

### DISCUSSION THREAD

**Discussion by xinlinhe** [carla-simulator/carla#4387] [Q&A]
I want a view that follows the vehicle, but when I follow the code below, the vehicle jitter is obvious in the simulator window, so there must be something wrong with it.
my code is as follows: 
 _while True:
      timestamp = world.wait_for_tick(10).timestamp
      self.simulation_time = timestamp.elapsed_seconds;
      spectator = world.get_spectator()
      actorMain_trans = self.main_vehicle.get_transform()
      spectator.set_transform(get_transform(location, actorMain_trans.rotation.yaw - 90, 7.0))_

Can anyone help me? I would be very grateful.

**Comment by StevenCoral** [comment]
It's no an editor view per-se, but you can attach a carla camera as a child to the parent vehicle, and then stream the output to a pygame window instance using the python api (see the examples on how to do so). It will prevent the "vehicle-updates-then-spectator-updates" based shaking.
